src/test/java/org/jenkinsci/plugins/envinject/GlobalPropertiesTest.java

```diff
@@ -91,4 +100,63 @@ public void testGlobalPropertiesSetWORKSPACE() throws Exception {
 
         assertEquals(result_testGlobalVariableName, result_testJobVariableName);
     }
+
+    @Test
+    @Issue("JENKINS-16316")
+    public void testChangeOfGlobalPropertyGetsRecognizedWhenWithoutJobPropertyAndRunOnSlaves() throws Exception {
```

Comment:
please avoid such long tests. Test should not be longer than 10-20 lines. Or it does something wrong